### PR TITLE
[GEP-17] Add owner check configuration to Etcd resource and controller

### PR DESCRIPTION
**Pull request by stoyanr**
**How to categorize this PR?**

/area control-plane-migration
/kind enhancement

**What this PR does / why we need it**:
Adds owner check parameters to the `Etcd` resource and adapts the controller to properly propagate them to the etcd statefulset as command line options to `etcd-backup-restore`. These options are being introduced with https://github.com/gardener/etcd-backup-restore/pull/383.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
* In `Draft` since it can only be merged once https://github.com/gardener/etcd-backup-restore/pull/383 has been merged and a version of `etcd-backup-restore` with the new options has been released.
* Temporarily sets a test image of `etcd-backup-restore` in `images.yaml` to enable testing.
* Temporarily fixes an issue in the etcd validation that is also fixed with #240, the corresponding commit will be removed when it's no longer needed.

**Release note**:

```other operator
It is now possible to specify owner check parameters in the Etcd resource.
```

### CODE REVIEW

File: api/v1alpha1/etcd_types.go [hunk 1]
```diff
@@ -147,6 +165,10 @@ type BackupSpec struct {
 	// EtcdSnapshotTimeout defines the timeout duration for etcd FullSnapshot operation
 	// +optional
 	EtcdSnapshotTimeout *metav1.Duration `json:"etcdSnapshotTimeout,omitempty"`
+	// OwnerCheck defines parameters related to checking if the seed is an owner
+	// of the shoot. The ownership can change during control plane migration.
+	// +optional
+	OwnerCheck *OwnerCheckSpec `json:"ownerCheck,omitempty"`
```
Comment (timuthy) [inline api/v1alpha1/etcd_types.go:171]: Does it make sense to add `OwnerCheck` to `EtcdSpec` directly? I expect that we'll need this information also when we need the[watchdog principle](https://github.com/gardener/gardener/pull/4638) in the Druid.

File: api/v1alpha1/etcd_types.go [hunk 2]
```diff
@@ -103,6 +103,24 @@ type CompressionSpec struct {
...
+	ID string `json:"id"`
+	// Interval is the time interval between owner checks.
+	// +optional
+	Interval *metav1.Duration `json:"interval,omitempty"`
+	// Timeout is the timeout for owner checks.
+	// +optional
+	Timeout *metav1.Duration `json:"timeout,omitempty"`
+	// DNSCacheTTL is the DNS cache TTL for owner checks.
+	// +optional
+	DNSCacheTTL *metav1.Duration `json:"dnsCacheTTL,omitempty"`
```
Comment (timuthy) [inline api/v1alpha1/etcd_types.go:121]: Aren't `Interval`, `Timeout` and `DNSCacheTTL` rather global configuration values (i.e. etcd-druid flags) for the controller?

File: api/v1alpha1/etcd_types.go [hunk 3]
```diff
@@ -103,6 +103,24 @@ type CompressionSpec struct {
 	Policy *CompressionPolicy `json:"policy,omitempty"`
 }
 
+// OwnerCheckSpec defines parameters related to checking if the seed is an owner
+// of the shoot. The ownership can change during control plane migration.
```
Comment (timuthy) [inline api/v1alpha1/etcd_types.go:107]: Can you define this in a general way, i.e. no Gardener architecture terms? etcd-druid can be used w/o Gardener and I also think the ownership check can be usable outside of the Gardener context as well, right?

File: charts/etcd/templates/etcd-statefulset.yaml [hunk 4]
```diff
@@ -190,6 +203,7 @@ spec:
         - --etcd-defrag-timeout={{ .Values.etcd.etcdDefragTimeout }}
 {{- end }}
         - --snapstore-temp-directory={{ .Values.backup.snapstoreTempDir }}
+        - --etcd-pid-command=ps ax -o pid,comm | grep 'etcd$' | awk '{print $1}' | { grep -Eo '[0-9]{1,}' || true; }
```
Comment (timuthy) [inline charts/etcd/templates/etcd-statefulset.yaml:206]: See https://github.com/gardener/etcd-backup-restore/pull/383#discussion_r728852543

File: controllers/etcd_controller.go [hunk 5]
```diff
@@ -1080,7 +1080,7 @@ func (r *EtcdReconciler) getMapFromEtcd(etcd *druidv1alpha1.Etcd) (map[string]in
 	}
 
 	backupValues := map[string]interface{}{
-		"pullPolicy":               corev1.PullIfNotPresent,
+		"pullPolicy":               corev1.PullAlways,
```
Comment (timuthy) [inline controllers/etcd_controller.go:1083]: I suspect this is only for testing?

File: charts/etcd/templates/etcd-statefulset.yaml (hunk 4 above)
Comment (stoyanr) [reply charts/etcd/templates/etcd-statefulset.yaml:206]: As agreed in the other PR, will try to use `go-ps` as suggested. There should still be a parameter I believe, for the etcd process name (defaulting to `etcd`).

Comment (timuthy) [reply charts/etcd/templates/etcd-statefulset.yaml:206]: Thanks a lot @stoyanr

**Comment by stoyanr** [comment]
@timuthy Addressed your comments, rebased on top of latest master, using the latest `etcd-backup-restore` release v0.14.0, switched the pull policy to `IfNotPresent`, and removed redundant temporary commits. I consider this PR ready to be reviewed / merged now.

**Review by timuthy — APPROVED** [review]
/lgtm

**Pull request merged**